### PR TITLE
Refactor RegExp Documentation: Fix Typos and Enhance Formatting;

### PR DESCRIPTION
**Pull request by A-Berezvay**
### Type of Change

- Adding a new entry
- Editing an existing entry (fixing a typo, bug, issues, etc)
- Updating the documentation

### Checklist

- [ x] All writings are my own.
- [ x] My entry follows the Codecademy Docs style guide.
- [ x] My changes generate no new warnings.
- [ x] I have performed a self-review of my own writing and code.
- [x ] I have checked my entry and corrected any misspellings.
- [ x] I have made corresponding changes to the documentation if needed.
- [ x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x ] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [ x] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

**Comment by A-Berezvay** [comment]
Great thank you! Let me know if anything require further assistance :)

**Comment by A-Berezvay** [comment]
Hi @dakshdeepHERE , could you please review this PR when you have a chance? Thanks!

File: content/javascript/concepts/regexp/regexp.md [hunk 1]
```diff
@@ -12,53 +12,51 @@ CatalogContent:
...
 
 A `RegExp` object can also have flags set along with a pattern to change how matches are performed.
 
 ## Syntax
 
-There are two methods of creating a `RegExp` object. The first method is literal notation using slashes to delimit the pattern, followed by any flags. The second method uses the `RegExp` constructor which takes the pattern as the first argument and any flags as the second.
+There are two ways to create a `RegExp` object:
 
+1. **Literal Notation**: Use slashes to delimit the pattern, followed by any flags.
 ```pseudo
```
Comment (avdhoottt) [inline content/javascript/concepts/regexp/regexp.md:24]: Use ` ```js ` for the codeblock.

File: content/javascript/concepts/regexp/regexp.md [hunk 2]
```diff
@@ -12,53 +12,51 @@ CatalogContent:
...
-// Using literal notation
 let re1 = /foo?/i;
+```
 
-// Using RegExp constructor
+2. **Constructor Function**: Use the `RegExp` constructor, passing the pattern as the first argument and any flags as the second. 
+```pseudo
```
Comment (avdhoottt) [inline content/javascript/concepts/regexp/regexp.md:29]: Use ` ```js ` for the codeblock.

**Review by avdhoottt — APPROVED** [review]
Hey @A-Berezvay, excellent entry! I have only one change to be made. Please do it as soon as possible. Thanks!!

**Review by Radhika-okhade — APPROVED** [review]

**Review by avdhoottt — APPROVED** [review]
LGTM!!

**Pull request merged**